setup1-cli-tools/src/public_key_extractor.rs

```diff
@@ -0,0 +1,61 @@
+use age::Decryptor;
+use anyhow::{anyhow, Result};
+use secrecy::{ExposeSecret, SecretString, SecretVec};
+use serde::Deserialize;
+use snarkos_toolkit::account::{Address, PrivateKey};
+use std::{fs, io::Read, str::FromStr};
+use structopt::StructOpt;
+use unic_langid::LanguageIdentifier;
+
+#[derive(Debug, StructOpt)]
+#[structopt(name = "Public key extractor")]
+struct Options {
+    #[structopt(long)]
+    path: String,
+}
+
+// Should be the same as the one from setup1-contributor/src/objects.rs
+// Copied here to reduce the compile time, which is
+// about 50% longer with setup1-contributor included
+#[derive(Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct AleoSetupKeys {
+    pub encrypted_seed: String,
+    pub encrypted_private_key: String,
+    pub address: String,
+}
+
+fn decrypt(passphrase: &SecretString, encrypted: &str) -> Result<SecretVec<u8>> {
+    let decoded = SecretVec::new(hex::decode(encrypted)?);
+    let decryptor = Decryptor::new(decoded.expose_secret().as_slice())?;
+    let default_language: LanguageIdentifier = "en-US".parse()?;
```

Comment:
No real reason, I don't know what's the best way to use age, but it's ok to me as far as it works and the tool generates the keys we need. Would you like me to move it to the main function?

Reply:
Oh wait, I think it was a mistake, it should actually go to read_private_key function, but somehow it ended up in decrypt function. Now I'm even more surprised because `decrypt` call happens after `age::cli_common::read_secret`, but it doesn't panic. Is it because the call of read_secret is lazy and everything is streamed on demand?

Nevertheless, I've moved language selection to main, thx for bringing this up!